binder/src/main/java/io/grpc/binder/ServerSecurityPolicy.java

```diff
@@ -44,16 +46,31 @@ private ServerSecurityPolicy(ImmutableMap<String, SecurityPolicy> perServicePoli
   /**
    * Return whether the given Android UID is authorized to access a particular service.
    *
-   * <b>IMPORTANT</b>: This method may block for extended periods of time.
+   * <p><b>IMPORTANT</b>: This method may block for extended periods of time.
    *
    * @param uid The Android UID to authenticate.
    * @param serviceName The name of the gRPC service being called.
+   * @deprecated Application code should not need to call this method.
    */
   @CheckReturnValue
+  @Deprecated
   public Status checkAuthorizationForService(int uid, String serviceName) {
     return perServicePolicies.getOrDefault(serviceName, defaultPolicy).checkAuthorization(uid);
   }
 
+  /**
+   * Return whether the given Android UID is authorized to access a particular service.
+   *
+   * @param uid The Android UID to authenticate.
+   * @param serviceName The name of the gRPC service being called.
+   */
+  @CheckReturnValue
+  ListenableFuture<Status> checkAuthorizationForServiceAsync(int uid, String serviceName) {
```

Comment:
Supporting `ListenableFuture<Status>` is actually kind of the main goal here, because I want make the framework support a client-provided `SecurityPolicy` that has a slow implementation without blocking the calling thread. An [earlier version](https://github.com/grpc/grpc-java/pull/10596/commits/9a94b59bd4ae1ce55ba38cb9288d267b0ae1cd12) of this PR may provide more context on what I'm aiming for, though @markb74 recommended splitting it to make the review easier.

Also note that this method is package-private. The bulk of the documentation is coming for the public `AsyncSecurityPolicy`. I think there's more value in investing there,  but please let me know if you feel strongly about documenting this one too.

Reply:
I'm not arguing against ListenableFuture. I'm just pointing out that ListenableFutures can complete with any type of exception and the compiler/IDE doesn't help you formalize them with throws declarations or @throws javadoc. As a reader it really isn't clear what it would mean for the returned future to complete with io.grpc.Status.PERMISSION_DENIED.asException() vs completing successfully with Status.PERMISSION_DENIED. 

But thanks for pointing out that this is not actually a public method. Up to you then whether you think formal javadoc is necessary.

Reply:
Thanks for that addition, although that part is kind of implied by the lack of throws clauses in the declaration.

What I was really asking for is to explain what to expect when it comes to the returned future completing with an exception. How about something like "The returned future is not expected to complete with an Exception -- if it does, that should be interpreted as a failure to make an authorization decision, not the authorization decision itself."

Reply:
Updated.